log/logger.go

```diff
@@ -0,0 +1,22 @@
+package log
+
+import (
+	"io"
+)
+
+// Option modifies the logger behavior
+type Option func(interface{})
+
+// Logger is a convenient interface to use provided loggers
+// either use it as it is or implement your own interface where
+// the logging implementations are used
+type Logger interface {
+	D(msg string, args ...interface{})
```

Comment:
Shall we use the full name instead of one letter word to improve readability?

Reply:
Sure, let's use the full name.